### DISCUSSION TITLE
Any plans to make it a postgres extension?

### DISCUSSION THREAD

**Discussion by mercurial-moon** [janbjorge/pgqueuer#143] [Ideas]
Hi are there any plans to make `pgqueuer` into a postgres extension. That way we could directly talk to it with sql, and it would be language agnostic.
Is there a way to do it currently? 
So we can use it from other programming languages.

**Comment by janbjorge** [comment]
Hi there,

Thanks for reaching out with your idea!

Currently, pgqueuer is not implemented as a PostgreSQL extension, so it isn't possible to interact with it directly via postgres. However, I really like your suggestion since a PostgreSQL extension could indeed make pgqueuer more flexible and language agnostic. 

However there is quite a bit of plumbing around the SQL part to make this work, so I'm a bit unsure how helpful it would be to others, but I could maybe split out the PostgreSQL part into an extension at some point if there is enough interest. If i where to make it, how would it help you?

**Reply by mercurial-moon** [reply]
Hi,
Thanks for the heads up.
I actually got the idea from `pgmq`. I was planning to write my own job queue on postgres using something similar to what you have done. But then I found your library only limitation being its in python and it  doesn't play well with the environment I'm currently using. 
I'm sure there's quite a bit of work involved to convert it to a pg extension.
I just provided the info so that it could make your library more rounded and open it up for use by more users.
Thanks again.
